primitives/plutus-data/src/lib.rs

```diff
@@ -15,12 +15,68 @@ pub struct DataDecodingError {
 type DecodingResult<T> = std::result::Result<T, DataDecodingError>;
 
 pub trait PlutusDataExtensions {
+	fn as_u32(self) -> Option<u32>;
 	fn as_u16(self) -> Option<u16>;
 }
 
 impl PlutusDataExtensions for cardano_serialization_lib::PlutusData {
+	fn as_u32(self) -> Option<u32> {
+		u32::try_from(self.as_integer()?.as_u64()?).ok()
+	}
 	fn as_u16(self) -> Option<u16> {
-		u16::try_from(u32::try_from(self.as_integer()?.as_u64()?).ok()?).ok()
+		u16::try_from(self.as_u32()?).ok()
+	}
+}
+
+/// Trait that provides decoding of versioned generic plutus data.
+///
+/// Versioned generic plutus data contain a version number and two data sections:
+/// - immutable section - the proper datum with stable schema, read and validated by smart contracts
```

Comment:
The other part is as proper as this one. One is read/written by smart-contracts the other is not used.

Reply:
dropped the "proper"